dask_expr/io/io.py

```diff
@@ -147,6 +148,58 @@ def _tune_up(self, parent):
         return
 
 
+class FusedParquetIO(FusedIO):
+    _parameters = ["_expr"]
+
+    @functools.cached_property
+    def _name(self):
+        return (
+            funcname(type(self.operand("_expr"))).lower()
+            + "-fused-parq-"
+            + _tokenize_deterministic(*self.operands)
+        )
+
+    @staticmethod
+    def _load_multiple_files(frag_filters, columns, schema, index_name):
+        from dask_expr.io.parquet import ReadParquetPyarrowFS
+
+        # Note: Ideally we'd build a pyarrow fragment that points to multiple
+        # tables since pyarrow handles IO and CPU threading separately and we're
+        # just overcomitting threads here. However, this isn't exposed at time
+        # of writing
+        # TODO: We may want to use a gloal pool with sufficient threads since
+        # executing this task in a local executor could be a little too much
+        tables = (
+            ReadParquetPyarrowFS._fragment_to_table(
+                frag,
+                filter,
+                columns,
+                schema,
+            )
+            for frag, filter in frag_filters
+        )
+        table = pa.concat_tables(tables)
```

Comment:
We may need to deal with type promotion (`promote_options="permissive")` here if we want to accommodate varying types in the input files.

Reply:
See https://github.com/hendrikmakait/distributed/blob/c6ce3bb52d7c41437efc54b294dd38891fa89b97/distributed/shuffle/_arrow.py#L52-L62 for the corresponding code snippet from P2P.

Reply:
this all is guaranteed to use pyarrow>=15 so this compat things isn't necessary

Reply:
We'd still need the `promote_options` kwarg if I'm not mistaken.

Reply:
yes, just not the compat foo